### PR TITLE
Initial Osmosis support, Address support

### LINKED ISSUE #1672: Add Osmosis support
Osmosis is a Cosmos-ecosystem AMM, with IBC cross chain communication.

Links:
https://osmosis.zone/
https://github.com/osmosis-labs/osmosis

Address derivation:
BIP44 coin ID is 118, same as for Cosmos
Sample address: osmo18s0hdnsllgcclweu9aymw4ngktr2k0rkvn7jmn which corresponds to cosmos18s0hdnsllgcclweu9aymw4ngktr2k0rkygdzdp

Osmosis token:
Symbol OSMO, decimals 6, base unit uosmo

Explorer:
https://www.mintscan.io/osmosis

Chain:
Chain ID osmosis-1

Public nodes:
https://rpc-osmosis.keplr.app
https://osmosis.validator.network
these seems to be Tendermint API, Cosmos API not known yet

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Osmosis support.  New chain, Cosmos-based.  Address and OSMO transfer supported.  Swap is not handled in this PR.
Fixes #1672 .

## Testing instructions

Osmo unit tests.
Get some Osmo using Osmosis app, and create transfer TX to another address.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: registry.json [hunk 1]
```diff
@@ -1816,5 +1816,31 @@
...
+  },
+  {
+    "id": "osmosis",
+    "name": "Osmosis",
+    "displayName": "Osmosis",
+    "coinId": 10000118,
+    "symbol": "OSMO",
+    "decimals": 6,
+    "blockchain": "Cosmos",
+    "derivationPath": "m/44'/118'/0'/0/0",
```
Comment (vikmeup) [inline registry.json:1828]: is this what `Keplr` wallet uses too?

Comment (catenocrypt) [reply registry.json:1828]: Yes

**Comment by catenocrypt** [comment]
Complete; address and coin transaction support; ready for review and merge.

**Comment by catenocrypt** [comment]
Kotlin test is missing, logo needs to be added to asseets (back to draft)

**Comment by catenocrypt** [comment]
Osmois logo in assets:  https://github.com/trustwallet/assets/pull/15589

**Comment by catenocrypt** [comment]
Address and OSMO transfer ready.  Tests complete, builds green.  Ready for review+merge.

**Review by hewigovens — APPROVED** [review]
🚀

**Review by vikmeup — APPROVED** [review]

**Pull request merged**